doc/HttpTransportAdapter.md

```diff
@@ -0,0 +1,223 @@
+# HTTP Transport Adapter
+
+The Azure SDK for C++ creates and sends HTTP requests using an HTTP Transport Adapter. This adapter provides a general abstraction for an HTTP stack (like libcurl, WinHTTP, etc) by defining the interface it needs to call that anyone can implement to customize which HTTP stack gets used.
+
+## Default HTTP Transport Adapter
+
+The Azure SDK for C++ provides two HTTP transport adapters as part of the azure-core package:
+- The libcurl Transport Adapter
+
+<center>
+
+| Supported Platforms  |
+| --- |
+| Linux |
+| Windows |
+| MacOS |
+| [Others](https://daniel.haxx.se/blog/wp-content/uploads/2020/02/slide-operating-systems.jpg) |
+
+</center>
+
+- The WinHTTP Transport Adapter
+
+<center>
+
+| Supported Platforms  |
+| --- |
+| Windows |
+
+</center>
+
+Azure SDK for C++ users can create and use their own transport adapter [by following the guidance below](#building-a-custom-http-transport-adapter).
+
+For the simplest case, when no specific configuration is set, the default transport adapter will be selected based on the OS. The WinHTTP transport adapter will be used for Windows-based systems (making libcurl an optional dependency). The libcurl transport adapter will be selected for any non-Windows-based system.
+
+## Building HTTP Transport Adapter
+
+The Azure SDK for C++ uses CMake options to define what HTTP transport adapters to build.
+
+You can see [CMake options](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#cmake-build-options) to learn about all the supported options and what is the specific option required for each HTTP transport adapter.
```

Comment:
Note to self, we need to add `BUILD_TRANSPORT_WINHTTP` to the CMake options table.

Reply:
#1300 assigned to you